test/framework/control_plane.go

```diff
@@ -34,84 +34,191 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-const (
-	// eventuallyInterval is the polling interval used by gomega's Eventually
-	eventuallyInterval = 10 * time.Second
+// Package globals used to populate default values.
+// Can be updated by consumer to change default values.
+var (
+	// DefaultTimeout is used for the default timeout for every `Eventually` call.
+	DefaultTimeout = 10 * time.Minute
+
+	// DefaultPollInterval is used as the default for how frequently `Eventually` polls.
+	DefaultPollInterval = 10 * time.Second
 )
 
-// ControlplaneClusterInput defines the necessary dependencies to run a multi-node control plane cluster.
-type ControlplaneClusterInput struct {
-	Management        ManagementCluster
-	Cluster           *clusterv1.Cluster
-	InfraCluster      runtime.Object
-	Nodes             []Node
-	MachineDeployment MachineDeployment
-	RelatedResources  []runtime.Object
-	CreateTimeout     time.Duration
-	DeleteTimeout     time.Duration
+// Interfaces to scope down client.Client
 
-	ControlPlane    *controlplanev1.KubeadmControlPlane
-	MachineTemplate runtime.Object
+// Getter can get resources.
+type Getter interface {
+	Get(ctx context.Context, key client.ObjectKey, obj runtime.Object) error
+}
+
+// Creator can creates resources.
+type Creator interface {
+	Create(ctx context.Context, obj runtime.Object, opts ...client.CreateOption) error
+}
+
+// Lister can lists resources.
+type Lister interface {
+	List(ctx context.Context, list runtime.Object, opts ...client.ListOption) error
+}
+
+// Deleter can delete resources.
+type Deleter interface {
+	Delete(ctx context.Context, obj runtime.Object, opts ...client.DeleteOption) error
+}
+
+// ListDeleter combines interfaces.
+type ListerDeleter interface {
+	Lister
+	Deleter
+}
+
+// Durations allow the customization of a test timeout and polling interval.
+// They can be customized for each test and/or the default can be changed by updating the package globals.
+type Durations struct {
+	Timeout      time.Duration
+	PollInterval time.Duration
 }
 
-// SetDefaults defaults the struct fields if necessary.
-func (input *ControlplaneClusterInput) SetDefaults() {
-	if input.CreateTimeout == 0 {
-		input.CreateTimeout = 10 * time.Minute
+// SetDefaults is a convenience for library consumers so they never have to specify values for this type.
+func (d *Durations) SetDefaults() {
+	if d.Timeout == 0 {
+		d.Timeout = DefaultTimeout
 	}
+	if d.PollInterval == 0 {
+		d.PollInterval = DefaultPollInterval
+	}
+}
 
-	if input.DeleteTimeout == 0 {
-		input.DeleteTimeout = 5 * time.Minute
+// CreateRelatedResourcesInput is the input type for CreateRelatedResources.
+type CreateRelatedResourcesInput struct {
+	Creator          Creator
+	RelatedResources []runtime.Object
+	Durations
+}
+
+// CreateRelatedResources is used to create runtime.Objects.
+func CreateRelatedResources(ctx context.Context, input CreateRelatedResourcesInput) {
```

Comment:
The current inputs limit the use of timeout and interval settings to just `time.Duration` values, I'm wondering if it would make sense to modify this to: `func CreateRelatedResources(ctx context.Context, input CreateRelatedResourcesInput, intervals ...interface{})` to allow us to forward them through to the `Eventually` call and not limit the types of values from what `Eventually` accepts.

Reply:
This makes sense to me. It keeps the spirit of Ginkgo and will allow for custom timeouts per test PLUS allows using the built-in global timeout setting functions to work as expected. Thanks for the feedback